yarn-project/sequencer-client/src/slasher/slasher_client.ts

```diff
@@ -0,0 +1,408 @@
+import {
+  type L2Block,
+  type L2BlockId,
+  type L2BlockSource,
+  L2BlockStream,
+  type L2BlockStreamEvent,
+  type L2Tips,
+} from '@aztec/circuit-types';
+import { INITIAL_L2_BLOCK_NUM } from '@aztec/circuits.js/constants';
+import { type L1ContractsConfig, type L1ReaderConfig, createEthereumChain } from '@aztec/ethereum';
+import { EthAddress } from '@aztec/foundation/eth-address';
+import { createLogger } from '@aztec/foundation/log';
+import { type AztecKVStore, type AztecMap, type AztecSingleton } from '@aztec/kv-store';
+import { SlashFactoryAbi } from '@aztec/l1-artifacts';
+import { type TelemetryClient, WithTracer } from '@aztec/telemetry-client';
+import { NoopTelemetryClient } from '@aztec/telemetry-client/noop';
+
+import {
+  type Chain,
+  type GetContractReturnType,
+  type HttpTransport,
+  type PublicClient,
+  createPublicClient,
+  getAddress,
+  getContract,
+  http,
+} from 'viem';
+
+/**
+ * Enum defining the possible states of the Slasher client.
+ */
+export enum SlasherClientState {
+  IDLE,
+  SYNCHING,
+  RUNNING,
+  STOPPED,
+}
+
+/**
+ * The synchronization status of the Slasher client.
+ */
+export interface SlasherSyncState {
+  /**
+   * The current state of the slasher client.
+   */
+  state: SlasherClientState;
+  /**
+   * The block number that the slasher client is synced to.
+   */
+  syncedToL2Block: L2BlockId;
+}
+
+export interface SlasherConfig {
+  blockCheckIntervalMS: number;
+  blockRequestBatchSize: number;
+}
+
+type SlashEvent = {
+  epoch: bigint;
+  amount: bigint;
+  lifetime: bigint;
+};
+
+/**
+ * @notice A Hypomeiones slasher client implementation
+ *
+ * Hypomeiones: a class of individuals in ancient Sparta who were considered inferior or lesser citizens compared
+ * to the full Spartan citizens.
+ *
+ * The implementation here is less than ideal. It exists, not to be the end all be all, but to show that
+ * slashing can be done with this mechanism.
+ *
+ * The implementation is VERY brute in the sense that it only looks for pruned blocks and then tries to slash
+ * the full committee of that.
+ * If it sees a prune, it will mark the full epoch as "to be slashed" and the
+ *
+ * Also, it is not particularly smart around what it should if there were to be multiple slashing events.
+ *
+ * A few improvements:
+ * - Only vote on the proposal if it is possible to reach, e.g., if 6 votes are needed and only 4 slots are left don't vote.
+ * - Stop voting on a payload once it is processed.
+ * - Only vote on the proposal if it have not already been executed
+ *  - Caveat, we need to fully decide if it is acceptable to have the same payload address multiple times. In the current
```

Comment:
Interesting feature: it would decrease the overhead needed to slash someone, so maybe fewer petty crimes?

We could mitigate the "multiple slash" by adding a minimum delay.

Regardless, I default to awaiting user demand.